axelor-base/src/main/java/com/axelor/apps/base/service/BarcodeGeneratorServiceImpl.java

```diff
@@ -357,4 +357,72 @@ public String addCharPaddingBits(int barcodeLength, String serialno, BarcodeForm
         barcodeFormat,
         barcodeLength);
   }
+
+  @Override
+  public boolean checkSerialNumberConsistency(
+      String serialno, BarcodeTypeConfig barcodeTypeConfig, boolean isPadding)
+      throws AxelorException {
+    if (serialno != null && barcodeTypeConfig != null) {
+      BarcodeFormat barcodeFormat = null;
```

Comment:
I think it would be a small improvement to not initialize it, and let the compiler checks if the variable is set for every possibility